cilium/bpf_metadata.cc

```diff
@@ -78,6 +79,8 @@ class BpfMetadataConfigFactory : public NamedListenerFilterConfigFactory {
     uint32_t mark = (config->is_ingress_) ? 0x0A00 : 0x0B00;
     options->push_back(std::make_shared<Cilium::SocketMarkOption>(mark, 0));
 
+    options->push_back(std::make_shared<Cilium::IpTransparentSocketOption>());
```

Comment:
should this be conditional on`addPrivilegedSocketOptions` as well?

Reply:
this code path isn't executed during tests - hence no need to add the condition.